pilot/pkg/xds/cds_test.go

```diff
@@ -170,36 +172,35 @@ func TestSAN(t *testing.T) {
 			name:    "Kubernetes service and EDS ServiceEntry",
 			objs:    []runtime.Object{service, pod, endpoint},
 			configs: []config.Config{dr, seEDS},
-			// The ServiceEntry rule will "win" the pushContext.ServiceAccounts.
-			// However, the Service will be processed first into a cluster. Since its not external, we do not add the SANs automatically
-			sans: nil,
+			// The ServiceEntry rule will "win" the conflict since it is in the proxy namespace.
+			sans: []string{"se-top", "spiffe://cluster.local/ns/test/sa/se-endpoint"},
 		},
 		{
 			name:    "Kubernetes service and NONE ServiceEntry",
 			objs:    []runtime.Object{service, pod, endpoint},
 			configs: []config.Config{dr, seNONE},
-			// Service properly sets SAN. Since it's not external, we do not add the SANs automatically though
-			sans: nil,
+			// The ServiceEntry rule will "win" the conflict since it is in the proxy namespace.
+			sans: []string{"se-top"},
 		},
 		{
 			name:    "Kubernetes service and EDS ServiceEntry ISTIO_MUTUAL",
 			objs:    []runtime.Object{service, pod, endpoint},
 			configs: []config.Config{drIstioMTLS, seEDS},
-			// The Service has precedence, so its cluster will be used
-			sans: []string{"spiffe://cluster.local/ns/default/sa/pod"},
+			// The ServiceEntry rule will "win" the conflict since it is in the proxy namespace.
+			sans: []string{"se-top", "spiffe://cluster.local/ns/test/sa/se-endpoint"},
 		},
 		{
 			name:    "Kubernetes service and NONE ServiceEntry ISTIO_MUTUAL",
 			objs:    []runtime.Object{service, pod, endpoint},
 			configs: []config.Config{drIstioMTLS, seNONE},
-			// The Service has precedence, so its cluster will be used
-			sans: []string{"spiffe://cluster.local/ns/default/sa/pod"},
+			// The ServiceEntry rule will "win" the conflict since it is in the proxy namespace.
+			sans: []string{"se-top"},
 		},
 		{
 			name:    "NONE ServiceEntry ISTIO_MUTUAL",
 			configs: []config.Config{drIstioMTLS, seNONE},
 			// Totally broken; service level ServiceAccount are ignored.
-			sans: []string{"custom"},
+			sans: []string{"se-top"},
```

Comment:
This has no real change; I changed the test input to be consistently using `se-top` as the SE SAN